build.sbt

```diff
@@ -2,379 +2,402 @@ import java.io._
 lazy val ScalaVersions = Seq("2.11.8", "2.12.1")
 lazy val LibrarySeries = "1.7.0"
-lazy val LibraryVersion = sys.props.getOrElse("scalameta.version", computePreReleaseVersion(LibrarySeries))
+lazy val LibraryVersion =
+  sys.props.getOrElse("scalameta.version", computePreReleaseVersion(LibrarySeries))
 
 // ==========================================
 // Projects
 // ==========================================
 
-lazy val scalametaRoot = Project(
-  id = "scalametaRoot",
-  base = file(".")
-) settings (
-  sharedSettings,
-  noPublish,
-  unidocSettings,
-  commands += Command.command("ci-fast") { state =>
-    // NOTE. The so/wow/such/very commands are from sbt-doge and are used to
-    // run commands with the correct scalaVersion in each project.
-    "so scalametaRoot/test" ::
-    "so contrib/test" ::
-    // slow tests below
-    "much doc" ::
-    state
-  },
-  commands += Command.command("ci-slow") { state =>
-    "very scalahost/test:runMain scala.meta.tests.scalahost.converters.LotsOfProjects" ::
-    "such testkit/test:runMain scala.meta.testkit.ScalametaParserPropertyTest" ::
-    "scalahostSbt/test" ::
-    state
-  },
-  commands += Command.command("ci-publish") { state =>
-    "very publish" ::
+name := "scalametaRoot"
+sharedSettings
+noPublish
+unidocSettings
+commands += Command.command("ci-fast") { state =>
+  if (isScalaJS) "scalametaJS/test" :: state
+  else "test" :: "doc" :: state
+}
+commands += Command.command("ci-slow") { state =>
+  if (isScalaJS) state // do nothing
+  else {
+    "scalahost/test:runMain scala.meta.tests.scalahost.converters.LotsOfProjects" ::
+      "testkit/test:runMain scala.meta.testkit.ScalametaParserPropertyTest" ::
+      "scalahostSbt/test" ::
       state
-  },
-  packagedArtifacts := Map.empty,
-  unidocProjectFilter in (ScalaUnidoc, unidoc) := inAnyProject,
-  aggregate in test := false,
-  test := {
-    val runScalametaTests = (test in scalameta in Test).value
-    val runScalahostTests = (test in scalahost in Test).value
-    val runBenchmarkTests = (test in benchmarks in Test).value
-    val runContribTests = (test in contrib in Test).value
-    val runDocs = (run in readme in Compile).toTask(" --validate").value
-  },
-  console := (console in scalameta in Compile).value
-) aggregate (
-  benchmarks,
-  common,
-  contrib,
-  dialects,
-  inline,
-  inputs,
-  parsers,
-  quasiquotes,
-  scalahost,
-  scalahostSbt,
-  scalameta,
-  semantic,
-  testkit,
-  tokenizers,
-  tokens,
-  transversers,
-  trees
-)
+  }
+}
+commands += Command.command("ci-publish") { state =>
+  for {
+    v <- sys.env.get("SCALA_VERSION")
+    t <- sys.env.get("TEST")
+    sjs <- sys.env.get("SCALA_VERSION")
+    if v == "2.11.8" && t == "ci-fast" && sjs == "false"
+  } {
+    RunSbtCommand("very publish")
+  }
+  state
+}
+packagedArtifacts := Map.empty
+unidocProjectFilter.in(ScalaUnidoc, unidoc) := inAnyProject
+aggregate.in(test) := false
+test := {
+  // `test` by default only runs on the JVM
```

Comment:
Absolutely. For example, if you use `java.io.File`.

However, I left JS out because it's slow, see https://github.com/scala-js/scala-js/issues/2827 Sometimes, incremental can take 3-4 minutes to link before running. The cause seems to be that the Scala.js linker struggles with inlining the thousand `test` method calls to Scalatest. I suggest we leave this for the CI server to catch.